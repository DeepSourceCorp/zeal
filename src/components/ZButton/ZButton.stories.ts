import ZButton from './ZButton.vue'

import { action } from '@storybook/addon-actions'

import '../../assets/css/tailwind.css'
import '../../assets/css/typography.css'
import '../../assets/css/layout.css'

export default {
  title: 'Zeal/Button',
  component: ZButton,
  excludeStories: /.*Data$/
}

export const DefaultWithoutText = () => ({
  components: { ZButton },
  template: '<z-button></z-button>'
})

export const ButtonWithText = () => ({
  components: { ZButton },
  template: '<z-button color="primary">Hello World</z-button>'
})

export const Colors = () => ({
  components: { ZButton },
  template: `<div>
        <z-button color="primary">Hello World</z-button>
        <z-button color="secondary">Hello World</z-button>
    </div>`
})

export const Sizes = () => ({
  components: { ZButton },
  template: `<div>
        <z-button size="small">Small Button</z-button>
        <z-button size="medium">Medium Button</z-button>
        <z-button size="large">Large Button</z-button>
        <z-button size="xlarge">Extra Large Button</z-button>
    </div>`
})

export const States = () => ({
  components: { ZButton },
  template: `<div>
        <z-button active>Active Button</z-button>
        <z-button disabled>Disabled Button</z-button>
        <z-button color="secondary" active>Active Secondary Button</z-button>
        <z-button color="secondary" disabled>Disabled Secondary Button</z-button>
    </div>`
})

export const FullWidthButton = () => ({
  components: { ZButton },
  template: `<div>
        <z-button full-width>Full Width Button</z-button>
    </div>`
})

export const DefaultLinkButton = () => ({
  components: { ZButton },
  template: `<div>
        <z-button color="link" type="link" to="https://deepsource.io/">Link Button</z-button>
    </div>`
})

export const ActiveLinkButton = () => ({
  components: { ZButton },
  template: `<div>
        <z-button color="link" type="link" to="https://deepsource.io/" active>Link Button</z-button>
    </div>`
})

export const DisabledLinkButton = () => ({
  components: { ZButton },
  template: `<div>
        <z-button color="link" type="link" to="https://deepsource.io/" disabled>Link Button</z-button>
    </div>`
})
