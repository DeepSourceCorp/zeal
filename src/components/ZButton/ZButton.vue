<template>
  <component
    :is="type === 'link' || to ? 'a' : 'button'"
    :href="to"
    class="z-btn"
    v-bind="$attrs"
    :disabled="disabled"
    v-on="$listeners"
    :class="[
      `${color && `z-btn--${color}`}`,
      `${size && `z-btn--${size}`}`,
      {
        'is-disabled': this.disabled !== false,
        'is-active': this.active !== false,
        'full-width': this.fullWidth !== false
      },
      customClasses,
      `${!color && !customClasses ? 'z-btn--primary' : ''}`
    ]"
    :type="type"
  >
    <slot>Click</slot>
  </component>
</template>

<script>
export default {
  name: 'ZButton',
  props: {
    color: {
      default: '',
      type: String
    },
    type: {
      default: 'button',
      type: String
    },
    customClasses: {
      default: '',
      type: String
    },
    size: {
      default: '',
      type: String
    },
    disabled: {
      default: false,
      type: Boolean
    },
    active: {
      default: false,
      type: Boolean
    },
    fullWidth: {
      default: false,
      type: Boolean
    },
    to: {
      default: '',
      type: String
    }
  },
  methods: {
    handleClick(event) {
      this.$emit('click', event)
    }
  }
}
</script>

<style lang="css" scoped>
/* Common Button Style */

.z-btn {
  @apply inline-flex items-center px-6 text-base font-medium border-transparent h-10 rounded-sm relative;
}

.is-disabled {
  @apply opacity-50 cursor-not-allowed;
}

.full-width {
  @apply w-full inline-block;
}

/* Button - Primary Style */

.z-btn--primary {
  @apply bg-juniper hover:bg-light_juniper text-ink-400;
}

.z-btn--primary.is-disabled {
  @apply hover:bg-juniper;
}

/* Button - Secondary Style */

.z-btn--secondary {
  @apply bg-ink-300 hover:bg-ink-200 text-vanilla-100 border border-slate;
}

.z-btn--secondary.is-disabled {
  @apply hover:bg-ink-300;
}

.z-btn--secondary.is-active {
  @apply bg-ink-100;
}

.z-btn--link {
  @apply font-normal text-juniper hover:underline;
}

.z-btn--link.is-active {
  @apply underline;
}

.z-btn--link.is-disabled {
  @apply text-slate no-underline;
}

/* Different Sizes of the Button*/
.z-btn--small {
  @apply h-8 text-xs;
}

.z-btn--medium {
  @apply h-10 text-base;
}

.z-btn--large {
  @apply h-12 text-lg;
}

.z-btn--xlarge {
  @apply h-16 text-lg;
}
</style>
