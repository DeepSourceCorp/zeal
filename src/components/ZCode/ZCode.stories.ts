import '../../assets/css/tailwind.css'
import '../../assets/css/typography.css'
import '../../assets/css/layout.css'
import ZCode from './ZCode.vue'

export default {
  title: 'Code',
  component: ZCode,
  excludeStories: /.*Data$/
}

export const DefaultCode = () => ({
  components: { ZCode },
  data() {
    return {
      content: `<div class='input-container'>
        <z-code :content="content"></z-code>           
    </div>`
    }
  },
  template: `<div class='input-container'>
        <z-code :content="content"></z-code>           
    </div>`
})
