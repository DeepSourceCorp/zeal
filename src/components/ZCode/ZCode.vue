<template>
  <div class="z-code" v-html="content"></div>
</template>
<script>
export default {
  name: 'ZCode',
  props: {
    content: {
      type: String
    }
  }
}
</script>
<style>
@import '../../assets/css/syntax_highlighter.css';
</style>
