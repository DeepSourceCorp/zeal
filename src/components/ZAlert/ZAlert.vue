<template>
  <div
    v-if="isVisible"
    class="px-4 rounded-md bg-opacity-10"
    :class="[bgColor, borderClass, dismissible ? 'py-2' : 'py-4']"
  >
    <div class="flex items-center" :class="{ 'py-px': !dismissible }">
      <div
        ref="text-container"
        class="flex-grow text-sm text-container"
        :class="[textColor, height > 32 ? 'py-2' : 'py-0']"
      >
        <slot />
      </div>

      <z-button
        v-if="dismissible"
        button-type="ghost"
        icon="x"
        :color="colors[type]"
        size="small"
        class="flex-shrink-0 hover:bg-opacity-0 place-self-start"
        @click="isVisible = false"
      />
    </div>

    <slot name="code-snippet" />
  </div>
</template>

<script lang="ts">
import Vue from 'vue'

import ZButton from '../ZButton/ZButton.vue'

export default Vue.extend({
  name: 'ZAlert',
  components: {
    ZButton
  },
  props: {
    type: {
      type: String,
      required: true,
      validator(val) {
        return ['info', 'warning', 'danger'].includes(val)
      }
    },
    dismissible: {
      type: Boolean,
      required: false,
      default: false
    },
    border: {
      type: String,
      required: false,
      validator(val) {
        return ['top', 'right', 'bottom', 'left', undefined].includes(val)
      },
      default: undefined
    }
  },
  data: function () {
    return {
      isVisible: true,
      height: 0,
      colors: {
        info: 'robin',
        warning: 'honey',
        danger: 'cherry'
      },
      observer: {} as ResizeObserver
    }
  },

  /**
   * Mouted hook for Vue component
   *
   * @returns {void}
   */
  mounted() {
    // Initialize the observer on mount
    this.initObserver()
  },

  /**
   * beforeDestroy hook for Vue component
   *
   * @returns {void}
   */
  beforeDestroy() {
    // Unobserve before destroy
    if (this.observer) {
      this.observer.unobserve(this.$refs['text-container'] as HTMLElement)
    }
  },
  methods: {
    /**
     * Callback function passed to ResizeObserver instance
     *
     * @returns {void}
     */
    onResize() {
      this.height = (this.$refs['text-container'] as HTMLElement).offsetHeight
    },

    /**
     * Initialize observer
     *
     * @returns {void}
     */
    initObserver() {
      const observer = new ResizeObserver(this.onResize)
      observer.observe(this.$refs['text-container'] as HTMLElement)
      this.observer = observer
    }
  },
  computed: {
    bgColor(): string {
      const bgColors = {
        info: 'bg-robin',
        warning: 'bg-honey',
        danger: 'bg-cherry'
      }
      return bgColors[this.type]
    },
    borderClass(): string | string[] {
      if (!this.border) {
        return ''
      }
      const borderTypes = {
        top: 'border-t',
        right: 'border-r',
        bottom: 'border-b',
        left: 'border-l'
      }
      const borderColors = {
        info: 'border-robin',
        warning: 'border-honey',
        danger: 'border-cherry'
      }
      return [borderTypes[this.border], borderColors[this.type], 'border-opacity-20']
    },
    textColor(): string {
      const textColors = {
        info: 'text-robin-400',
        warning: 'text-honey-400',
        danger: 'text-cherry-400'
      }
      return textColors[this.type]
    }
  }
})
</script>

<style scoped>
.text-container::first-line {
  line-height: 1;
}
</style>
