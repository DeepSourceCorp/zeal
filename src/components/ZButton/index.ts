import ZButton from './ZButton.vue';

export { ZButton };
