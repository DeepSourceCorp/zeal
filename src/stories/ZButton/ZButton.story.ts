import ZButton from '../../components/ZButton/ZButton.vue'
import { Meta, Story } from '@storybook/addon-docs/blocks'
import { component } from 'vue/types/umd'

export const ButtonWithText = () => ({
  components: { ZButton },
  template: '<z-button>Hello Button</button>',
  title: 'ZButton'
})

ButtonWithText.story = {
  name: 'Button with Text'
}
