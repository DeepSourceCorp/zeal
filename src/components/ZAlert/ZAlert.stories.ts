import Vue from 'vue'
import ZAlert from '../ZAlert/ZAlert.vue'
import ZButton from '../ZButton/ZButton.vue'
import ZCode from '../ZCode/ZCode.vue'

export default {
  title: 'Alert',
  component: ZAlert
}

/**
 * UI states representing the various Alert component types
 *
 * @returns {ExtendedVue}
 */
export const AlertTypes = () => {
  return Vue.extend({
    components: { ZAlert },
    data: function() {
      return {
        types: ['info', 'warning', 'danger']
      }
    },
    template: `
      <div class='text-vanilla-100 space-y-4'>
        <div v-for="type in types" :key="type">
          <z-alert :type="type">
            <p class="flex items-center h-8">
              Alert of the type {{ type }}
            </p>
          </z-alert>
        </div>
      </div>
    `
  })
}

/**
 * UI states representing dismissible Alert component
 *
 * @returns {ExtendedVue}
 */
export const DismissibleAlert = () => {
  return Vue.extend({
    components: { ZAlert },
    data: function() {
      return {
        types: ['info', 'warning', 'danger']
      }
    },
    template: `
      <div class='text-vanilla-100 space-y-4'>
        <div v-for="type in types" :key="type">
          <z-alert :type="type" :dismissible="true">
            <p class="flex items-center h-8">
              Alert of the type {{ type }}
            </p>
          </z-alert>
        </div>
      </div>
    `
  })
}

/**
 * UI states representing dismissible Alert component with controls
 *
 * @returns {ExtendedVue}
 */
export const DismissibleAlertWithControls = () => {
  return Vue.extend({
    components: { ZAlert, ZButton },
    data: function() {
      return {
        colors: {
          info: 'bg-robin',
          warning: 'bg-honey',
          danger: 'bg-cherry'
        },
        types: ['info', 'warning', 'danger']
      }
    },
    template: `
      <div class='text-vanilla-100 space-y-4'>
        <div v-for="type in types" :key="type">
          <z-alert :type="type" :dismissible="true">
            <div class="md:hidden flex flex-col space-y-4 mr-2">
              <p>Alert of the type {{ type }}</p>
              <div class="inline-flex space-x-2">
              <z-button button-type="ghost" color="vanilla-100" size="small" :class="[colors[type], 'bg-opacity-40']">
                Button 1
              </z-button>
              <z-button button-type="ghost" color="vanilla-100" size="small" :class="[colors[type], 'bg-opacity-40']">
                Button 2
              </z-button>
              </div>
            </div>
    
            <div class="hidden md:flex items-center justify-between mr-2">
              Alert of the type {{ type }}
              <z-button button-type="ghost" color="vanilla-100" size="small" :class="[colors[type], 'bg-opacity-40']">
                Default button
              </z-button>
            </div>
          </z-alert>
        </div>
      </div>
    `
  })
}

/**
 * UI states representing dismissible Alert component with controls and code snippet
 *
 * @returns {ExtendedVue}
 */
export const DismissibleAlertWithCodeSnippet = () => {
  return Vue.extend({
    components: { ZAlert, ZButton, ZCode },
    data: function() {
      return {
        colors: {
          info: 'bg-robin',
          warning: 'bg-honey',
          danger: 'bg-cherry'
        },
        content: `<div class="highlight"><pre><span class="ln">2</span>\n<span class="ln">3</span>\n<span class="ln">4</span><span class="nd">@dramatiq.actor</span>\n<span class="hl"><span class="ln">5</span><span class="k">def</span> <span class="nf">example</span><span class="p">(</span><span class="p">)</span><span class="p">:</span></span><span class="ln">6</span>    <span class="k">pass</span>\n</pre></div>`,
        types: ['info', 'warning', 'danger']
      }
    },
    template: `
      <div class='text-vanilla-100 space-y-4'>
        <div v-for="type in types" :key="type">
          <z-alert :type="type" :dismissible="true">
            <div class="md:hidden mr-2">
              Alert of the type {{ type }}
            </div>

            <div class="hidden md:flex items-center justify-between mr-2">
              <p> Alert of the type {{ type }} </p>
              <z-button button-type="ghost" color="vanilla-100" size="small" :class="[colors[type], 'bg-opacity-40']">
                Default button
              </z-button>
            </div>

            <template #code-snippet>
              <z-code :content="content" />

              <div class="md:hidden inline-flex space-x-2 mt-4">
                <z-button button-type="ghost" color="vanilla-100" size="small" :class="[colors[type], 'bg-opacity-40']">
                  Button 1
                </z-button>
                <z-button button-type="ghost" color="vanilla-100" size="small" :class="[colors[type], 'bg-opacity-40']">
                  Button 2
                </z-button>
              </div>
            </template>
        </z-alert>
      </div>
    `
  })
}

/**
 * UI states representing dismissible Alert component with borders
 *
 * @returns {ExtendedVue}
 */
export const DismissibleAlertWithBorders = () => {
  return Vue.extend({
    components: { ZAlert },
    data: function() {
      return {
        borderTypes: ['top', 'right', 'bottom', 'left']
      }
    },
    template: `
      <div class='text-vanilla-100 space-y-4'>
        <div v-for="type in borderTypes" :key="type">
          <z-alert type="danger" :dismissible="true" :border="type">
            <p class="flex items-center h-8">
              Alert of the type danger with border-{{ type }}
            </p>
          </z-alert>
        </div>
      </div>
    `
  })
}
