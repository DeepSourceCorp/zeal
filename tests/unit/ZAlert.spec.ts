import { createLocalVue, mount } from '@vue/test-utils'
import ZAlert from '../../src/components/ZAlert'
import ZButton from '../../src/components/ZButton'
import ZCode from '../../src/components/ZCode'

/**
 * Helper to mount the component
 *
 * @param {Object} options={}
 * @returns {Wrapper<Vue>}
 */
const mountFunction = (options = {}) => {
  const localVue = createLocalVue()
  return mount(ZAlert, {
    localVue,
    ...options
  })
}

describe('Alert component', () => {
  it('renders an alert widget with the given text', () => {
    expect(
      mountFunction({
        propsData: {
          type: 'danger'
        },
        slots: {
          default: 'Alert of the type danger'
        }
      }).html()
    ).toMatchSnapshot()
  })

  it('renders a dismissible alert widget', () => {
    expect(
      mountFunction({
        propsData: {
          type: 'danger',
          dismissible: true
        },
        slots: {
          default: 'Alert of the type danger which is dismissible'
        }
      }).html()
    ).toMatchSnapshot()
  })

  it('renders a dismissible alert widget with controls', () => {
    expect(
      mountFunction({
        components: {
          ZButton
        },
        propsData: {
          type: 'danger',
          dismissible: true
        },
        slots: {
          default: `<div class="hidden md:flex items-center justify-between mr-2">
          Alert of the type danger with controls
          <z-button button-type="ghost" color="vanilla-100" size="small" class="bg-cherry bg-opacity-40">
            Default button
          </z-button>
        </div>`
        }
      }).html()
    ).toMatchSnapshot()
  })

  it('renders a dismissible alert widget with code snippet', () => {
    expect(
      mount({
        template: `<z-alert type="danger" :dismissible="true">
            <div>
              <p> Alert of the type danger with code snippet </p>
              <z-button button-type="ghost" color="vanilla-100" size="small" class="bg-cherry bg-opacity-40">
                Default button
              </z-button>
            </div>

            <template #code-snippet>
              <z-code :content="content" />
            </template>
          </z-alert>`,
        components: {
          ZAlert,
          ZButton,
          ZCode
        },
        data() {
          return {
            content: `<z-code content="<div class="highlight"><pre><span class="ln">2</span>\n<span class="ln">3</span>\n<span class="ln">4</span><span class="nd">@dramatiq.actor</span>\n<span class="hl"><span class="ln">5</span><span class="k">def</span> <span class="nf">example</span><span class="p">(</span><span class="p">)</span><span class="p">:</span></span><span class="ln">6</span>    <span class="k">pass</span>\n</pre></div>" />`
          }
        }
      }).html()
    ).toMatchSnapshot()
  })
})
