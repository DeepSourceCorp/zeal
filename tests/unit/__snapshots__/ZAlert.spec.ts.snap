// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Alert component renders a dismissible alert widget 1`] = `
<div class="px-4 rounded-md bg-opacity-10 bg-cherry py-2">
  <div class="flex items-center">
    <div class="flex-grow text-sm text-cherry-400 -mb-px">Alert of the type danger which is dismissible</div> <button href="#" type="button" class="inline-flex items-center font-medium transition-colors duration-300 ease-in-out rounded-sm z-btn focus:outline-none whitespace-nowrap flex-shrink-0 hover:bg-opacity-0 place-self-start text-cherry justify-center z-btn--cherry p-0 h-8 w-8 font-normal hover:bg-ink-200"><svg viewBox="0 0 24 24" stroke-linecap="round" stroke-linejoin="round" fill="none" class="z-icon w-4 h-4 z-icon--x text-cherry">
        <line x1="18" y1="6" x2="6" y2="18"></line>
        <line x1="6" y1="6" x2="18" y2="18"></line>
      </svg> </button>
  </div>
</div>
`;

exports[`Alert component renders a dismissible alert widget with code snippet 1`] = `
<div class="px-4 rounded-md bg-opacity-10 bg-cherry py-2">
  <div class="flex items-center">
    <div class="flex-grow text-sm text-cherry-400 -mb-px">
      <div>
        <p> Alert of the type danger with code snippet </p> <button href="#" type="button" class="inline-flex items-center font-medium transition-colors duration-300 ease-in-out rounded-sm z-btn focus:outline-none whitespace-nowrap bg-cherry bg-opacity-40 text-vanilla-100 justify-center z-btn--vanilla-100 h-8 px-4 py-1 text-xs space-x-1 leading-loose font-normal hover:bg-ink-200">
          <!---->
          Default button
        </button>
      </div>
    </div> <button href="#" type="button" class="inline-flex items-center font-medium transition-colors duration-300 ease-in-out rounded-sm z-btn focus:outline-none whitespace-nowrap flex-shrink-0 hover:bg-opacity-0 place-self-start text-cherry justify-center z-btn--cherry p-0 h-8 w-8 font-normal hover:bg-ink-200"><svg viewBox="0 0 24 24" stroke-linecap="round" stroke-linejoin="round" fill="none" class="z-icon w-4 h-4 z-icon--x text-cherry">
        <line x1="18" y1="6" x2="6" y2="18"></line>
        <line x1="6" y1="6" x2="18" y2="18"></line>
      </svg> </button>
  </div>
  <div class="z-code">
    <z-code content="<div class=" highlight"=""><pre><span class="ln">2</span>
<span class="ln">3</span>
<span class="ln">4</span><span class="nd">@dramatiq.actor</span>
<span class="hl"><span class="ln">5</span><span class="k">def</span> <span class="nf">example</span><span class="p">(</span><span class="p">)</span><span class="p">:</span></span><span class="ln">6</span>    <span class="k">pass</span>
</pre>" /&gt;</z-code>
  </div>
</div>
`;

exports[`Alert component renders a dismissible alert widget with controls 1`] = `
<div class="px-4 rounded-md bg-opacity-10 bg-cherry py-2">
  <div class="flex items-center">
    <div class="flex-grow text-sm text-cherry-400 -mb-px">
      <div class="hidden md:flex items-center justify-between mr-2">
        Alert of the type danger with controls
        <button href="#" type="button" class="inline-flex items-center font-medium transition-colors duration-300 ease-in-out rounded-sm z-btn focus:outline-none whitespace-nowrap bg-cherry bg-opacity-40 text-vanilla-100 justify-center z-btn--vanilla-100 h-8 px-4 py-1 text-xs space-x-1 leading-loose font-normal hover:bg-ink-200">
          <!---->
          Default button
        </button>
      </div>
    </div> <button href="#" type="button" class="inline-flex items-center font-medium transition-colors duration-300 ease-in-out rounded-sm z-btn focus:outline-none whitespace-nowrap flex-shrink-0 hover:bg-opacity-0 place-self-start text-cherry justify-center z-btn--cherry p-0 h-8 w-8 font-normal hover:bg-ink-200"><svg viewBox="0 0 24 24" stroke-linecap="round" stroke-linejoin="round" fill="none" class="z-icon w-4 h-4 z-icon--x text-cherry">
        <line x1="18" y1="6" x2="6" y2="18"></line>
        <line x1="6" y1="6" x2="18" y2="18"></line>
      </svg> </button>
  </div>
</div>
`;

exports[`Alert component renders an alert widget with the given text 1`] = `
<div class="px-4 rounded-md bg-opacity-10 bg-cherry py-3.5">
  <div class="flex items-center">
    <div class="flex-grow text-sm text-cherry-400 -mb-px">Alert of the type danger</div>
    <!---->
  </div>
</div>
`;
